### DISCUSSION TITLE
CTT setup X509UserIdentity certs

### DISCUSSION THREAD

**Discussion by johannwesely** [eclipse-milo/milo#1283] [General]
Hi we are starting to test our milo server with ctt (ConformanceTestTool) i need to know where i have to copy the X509UserIdentity_PKI certs ? The ApplicationInstance_PKI certs i copied to C:\Users\...\AppData\Local\Temp\security\pki\trusted. Should I Copy the X509UserIdentity_PKI certs there as well ?

**Comment by kevinherron** [comment]
You can copy them wherever you want, the SDK support for X509 identities is limited to the `X509Validator` implementation, which requires you to provide a `Predicate<X509Certificate>` - how you validate that certificate and where you might store them is up to your implementation.

**Reply by kevinherron** [reply]
@johannwesely I believe what's recommended is to use a similar structure to the `pki` folder that exists for application instance certificates, so you might have both a `pki` and a `pki-user` or `user-pki` folder in the "security" directory.
